pkg/php/composer/parse.go

```diff
@@ -24,18 +34,75 @@ func NewParser() types.Parser {
 
 func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
 	var lockFile lockFile
-	decoder := json.NewDecoder(r)
-	err := decoder.Decode(&lockFile)
+	input, err := io.ReadAll(r)
 	if err != nil {
+		return nil, nil, xerrors.Errorf("read error: %w", err)
+	}
+	if err = jfather.Unmarshal(input, &lockFile); err != nil {
 		return nil, nil, xerrors.Errorf("decode error: %w", err)
 	}
 
-	var libs []types.Library
+	libs := map[string]types.Library{}
+	foundDeps := map[string][]string{}
 	for _, pkg := range lockFile.Packages {
-		libs = append(libs, types.Library{
-			Name:    pkg.Name,
-			Version: pkg.Version,
+		lib := types.Library{
+			ID:        utils.PackageID(pkg.Name, pkg.Version),
+			Name:      pkg.Name,
+			Version:   pkg.Version,
+			Indirect:  false, // composer.lock file doesn't have info about Direct/Indirect deps. Will think that all dependencies are Direct
+			License:   strings.Join(pkg.License, ", "),
+			Locations: []types.Location{{StartLine: pkg.StartLine, EndLine: pkg.EndLine}},
+		}
+		libs[lib.Name] = lib
+
+		var dependsOn []string
+		for depName := range pkg.Require {
+			// Require field includes required php version, skip this
+			// Also skip PHP extensions
+			if depName != "php" || strings.HasPrefix(depName, "ext") {
```

Comment:
I think it never reaches `strings.HasPrefix(depName, "ext")` due to `depName != "php"`. The comment says it skips PHP extensions, so I guess it should be as below.

```suggestion
			if depName != "php" && !strings.HasPrefix(depName, "ext") {
```

I prefer `A || B` rather than `!A && !B` to avoid this kind of bug, though.

```suggestion
			if depName == "php" || strings.HasPrefix(depName, "ext") {
				continue
			}
```